### PR TITLE
fix in getVirtualMachineCount

### PR DESCRIPTION
**Pull request by clguimanMSFT**
getVirtualMachineCount counts only the VMs deployed from the current Jenkins instance [FIXED JENKINS-41568]

### CODE REVIEW

File: src/main/java/com/microsoft/azure/vmagent/util/AzureUtil.java [hunk 1]
```diff
@@ -443,6 +443,10 @@ public boolean matches(final DeploymentTag rhs, long timeout) {
              return Math.abs(timestamp - rhs.timestamp) > timeout;
         }
 
+        public boolean isFromSameInstance(final DeploymentTag rhs) {
```
Comment (arroyc) [inline src/main/java/com/microsoft/azure/vmagent/util/AzureUtil.java]: anything better than rhs??  you can ignore too

File: src/test/java/com/microsoft/azure/vmagent/test/ITAzureVMManagementServiceDelegate.java [hunk 2]
```diff
@@ -289,7 +289,9 @@ public void getVirtualMachineCountTest() {
         try {
             Random rand = new Random();
             final int numberOfAgents = rand.nextInt(4) + 1;
+            final String vmName = "vmnotcounted";
```
Comment (arroyc) [inline src/test/java/com/microsoft/azure/vmagent/test/ITAzureVMManagementServiceDelegate.java]: should it be a constant?

Comment (arroyc) [inline src/test/java/com/microsoft/azure/vmagent/test/ITAzureVMManagementServiceDelegate.java]: ignore

Comment (clguimanMSFT) [inline src/test/java/com/microsoft/azure/vmagent/test/ITAzureVMManagementServiceDelegate.java]: it's a relevant constant only for this test

File: src/main/java/com/microsoft/azure/vmagent/util/AzureUtil.java (hunk 1 above)
Comment (clguimanMSFT) [inline src/main/java/com/microsoft/azure/vmagent/util/AzureUtil.java]: I'll rename

**Comment by mmitche** [comment]
@clguimanMSFT While this is the right change, it is a potentially confusing one.  If you deploy enough resources in another Jenkins instance, you could begin to hit quotas.  Without a "global limit" of some kind, or changing this plugin to detect quotas, Jenkins will continue to ask for VMs and will continue to fail.

**Comment by clguimanMSFT** [comment]
@mmitche but you can also hit the quota by just having a lot of VMs in other resource groups.
I guess there's no right answer :)

**Comment by mmitche** [comment]
Except perhaps by checking quota limits against what you're being asked to deploy.

**Comment by clguimanMSFT** [comment]
@rtyler you're the person that reported this. 
What's the least confusin meaning for the "Max Virtual Machine Limit" field?
1. Number of VMs this Jenkins instance can deploy in the specified resource group? This also means if another Jenkins instance or any other service is deploying VMs in that resource group we won't count those and your deployments will start failing when you Azure subscription quota is reached
2. Number of VMs the specified resource group can hold? This means if another service is deploying in your RG then you'll stop deploying agents when the quota is reached

**Comment by clguimanMSFT** [comment]
We just decided that this is the right behavior and we'll also update the label name to "Max Jenkins Agents Limit"

**Pull request merged**